pkg/util/eventqueue/eventqueue.go

```diff
@@ -0,0 +1,297 @@
+package eventqueue
```

Comment:
my understanding was probably flawed. Anyway, I'll see what I can do over the next couple of days - might be a stretch (for me) to get this wrapped up.